### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by rweyand**

### CODE REVIEW

**Comment by danpersa** [comment]
looks like travis has failed for this...

File: beard.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (xbsd-nikolay) [inline beard.iml]: IDEA related file shouldn't in the repo

File: src/main/scala/de/zalando/beard/filter/FilterResolver.scala [hunk 2]
```diff
@@ -10,12 +12,7 @@ import scala.collection.immutable.{Set, Seq, Map}
   */
 trait FilterResolver {
 
-  def registeredFilters = Seq(
-    LowercaseFilter(),
-    UppercaseFilter(),
-    DateFormatFilter(),
-    CapitalizeFilter()
-  )
+  def registeredFilters = Seq(LowercaseFilter(), UppercaseFilter(), DateFormatFilter(), CapitalizeFilter(), TranslationFilter())
```
Comment (xbsd-nikolay) [inline src/main/scala/de/zalando/beard/filter/FilterResolver.scala]: Better to keep filters on separate lines

File: beard.iml (hunk 1 above)
Comment (danpersa) [inline beard.iml]: true, *.iml should be in the `.gitignore` file

File: src/main/scala/de/zalando/beard/filter/FilterResolver.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/de/zalando/beard/filter/FilterResolver.scala]: true, looks better...

**Comment by rweyand** [comment]
Fixed CRs and build

**Comment by boopathi** [comment]
You should probably `rebase -i` and squash/fixup the commits

File: src/main/scala/de/zalando/beard/filter/Filter.scala [hunk 3]
```diff
@@ -19,6 +20,8 @@ case class WrongParameterTypeException(parameterName: String, paramterType: Stri
 case class TypeNotSupportedException(filterName: String, className: String) extends FilterException(filterName)
 case class FilterNotFound(filterName: String) extends FilterException(filterName)
 
+case class InputFormatException(filterName: String, message: String) extends FilterException(message)
```
Comment (boopathi) [inline src/main/scala/de/zalando/beard/filter/Filter.scala]: The filtername gets eaten up and is not shown in the console. It's better to prepend it to message.

``` scala
case class IFE(filterName, message) extends FilterException(s"${filterName} - ${message}");
```

File: src/test/scala/de/zalando/beard/filter/TranslationFilterTest.scala [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.scalatest.{FunSpec, Matchers}
+
+/**
+  * Created by rweyand on 3/23/16.
+  */
+class TranslationFilterTest extends FunSpec with Matchers{
+  describe("Translation filter") {
+    it("should  resolve locale and bundle when given correct arguments") {
+      val filter = TranslationFilter()
+      filter.apply("example.title", Map("bundle" -> "messages", "locale" -> "de")) shouldBe "Hallo"
```
Comment (danpersa) [inline src/test/scala/de/zalando/beard/filter/TranslationFilterTest.scala]: please use `should be` to be consistent...

**Pull request merged**